process_report/institute_list_models.py

```diff
@@ -0,0 +1,61 @@
+from typing import Annotated
+import datetime
+
+import pydantic
+import validators
+
+
+def parse_date(v: str) -> str:
+    try:
+        datetime.datetime.strptime(v, "%Y-%m")
+        return v
+    except ValueError:
+        raise ValueError(f"Invalid date string {v}. Must be in format YYYY-MM")
+
+
+def validate_domain(v: str) -> str:
+    if not validators.domain(
+        v, consider_tld=True
+    ):  # Ensures only TLDs allowed by IANA are valid
+        raise ValueError(f"Invalid domain {v}")
+    return v
+
+
+DateField = Annotated[str, pydantic.BeforeValidator(parse_date)]
+DomainField = Annotated[str, pydantic.AfterValidator(validate_domain)]
+
+
+class InstituteInfo(pydantic.BaseModel):
+    display_name: str
+    domains: list[DomainField]
+    mghpcc_partnership_start_date: DateField = None
```

Comment:
This is an invalid specification, because `None` is not a `DateField`. You need:

```suggestion
    mghpcc_partnership_start_date: DateField|None = None
```